pandas/core/frame.py

```diff
@@ -4606,7 +4602,34 @@ def query(self, expr: str, *, inplace: bool = False, **kwargs) -> DataFrame | No
             raise ValueError(msg)
         kwargs["level"] = kwargs.pop("level", 0) + 1
         kwargs["target"] = None
-        res = self.eval(expr, **kwargs)
+
+        # GH 59285
```

Comment:
there is already logic in eval to handling escaping special characters

needs to go there

Reply:
mmm and I can think of edge cases when the quick-and-dirty "tokenizing" with split wouldn't work. Let me make some changes